cmd/xmidt-agent/default-config.yaml

```diff
@@ -0,0 +1,184 @@
+#       # public jwt signing key(s)
+#       pems:
+#       - |
+#         -----BEGIN PUBLIC KEY-----
+#         MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAkZeQUMqDkMu/dQgTcycJ
+#         /rrgqfKQkoYFTMO7GnK87/OWiftHmtDJYPgktlcHheOPfkGln5ay1WJSitmtWNaH
+#         +RvBFK/ZsXpIuGm7R6wdSc7e6g9fTaSAfNX/+a8VxHUo58AinXxcq4LnHbuaGjEz
+#         jw77TLuZGyUuHiany8O8tc+DbnYKvRquScsccI6z/QwZKFtXUjJZ91hJ97zC8o7N
+#         Ae7n/Jg+Bs0uz9c1/bf/Jqbu6OidFbCr2FN42UupuAZ8DiPp2fWD5Q9qmp1ADk+V
+#         +TeZPxTCq/WB4dzSCd5v/FvFmO8tH6Ptkltij4pke7Dsi80TVRlcMDXAWxSFXOQV
+#         qwIDAQAB
+#         -----END PUBLIC KEY-----
+#       timeout: 10s
+websocket:
+  url_path:           "/api/v2/device"
+  # used if xmidt_service section is empty or xmdit_service connection fails
+  back_up_url:        "http://localhost:8080" 
+  fetch_url_timeout:  30s
+  ping_interval:      30s
+  ping_write_timeout:       90s
+  send_timeout:       90s
+  keep_alive_interval: 30s
+  http_client:
+    timeout: 30s
+    transport: 
+      idle_conn_timeout:       10s
+      tls_handshake_timeout:   10s
+      expect_continue_timeout: 1s
+  max_message_bytes: 262144 # 256 * 1024
+  #
+  #	This retry policy gives us a very good approximation of the prior
+  #	policy.  The important things about this policy are:
+  #
+  #	1. The backoff increases up to the max.
+  #	2. There is jitter that spreads the load so windows do not overlap.
+  #
+  #	iteration | parodus   | this implementation
+  #	----------+-----------+----------------
+  #	0         | 0-1s      |   0.666 -  1.333
+  #	1         | 1s-3s     |   1.333 -  2.666
+  #	2         | 3s-7s     |   2.666 -  5.333
+  #	3         | 7s-15s    |   5.333 -  10.666
+  #	4         | 15s-31s   |  10.666 -  21.333
+  #	5         | 31s-63s   |  21.333 -  42.666
+  #	6         | 63s-127s  |  42.666 -  85.333
+  #	7         | 127s-255s |  85.333 - 170.666
+  #	8         | 255s-511s | 170.666 - 341.333
+  #	9         | 255s-511s |           341.333
+  #	n         | 255s-511s |           341.333
+  #
+  retry_policy: 
+    interval: 1s
+    multiplier: 2.0
+    jitter: .33333333 #1.0 / 3.0
+    max_interval: 341333ms # 341*time.Second + 333*time.Millisecond
+lib_parodus: 
+  parodus_service_url: "tcp://127.0.0.1:6666"
+  keep_alive_interval: 30s
+  receive_timeout:    1s
+  send_timeout:       1s
+pubsub: 
+  publish_timeout: 5s
+logger:
+  encoder_config: 
+    time_key:        T
+    level_key:       L
+    name_key:        N
+    caller_key:      C
+    function_key:    "zapcore.OmitKey"
+    message_key:     M
+    stacktrace_key:  S
+    line_ending:     "zapcore.DefaultLineEnding"
+    encode_level:    capital
+    encode_time:     RFC3339Nano
+    encode_duration: string
+    encode_caller:   short
+  rotation: 
+    max_size:    1 #  1MB max/file
+    max_age:     30 # 30 days max
+    MaxBackups: 10 # max 10 files
+operational_state:
+  last_reboot_reason: sleepy
+  boot_time: "2024-02-28T01:04:27Z"
```

Comment:
should we have a default for `boot_time`? Maybe xmidt-agent can use `time.Now()` when `boot_time` is empty

Reply:
The problem with a boot time that is wrong but not clearly wrong is that it's really hard to disambiguate later.  If we leave it empty and then use the value that `time.Time{}` gives us then we know the value was not set.  I think this value is like `0001-01-01T00:00:00Z`.

Reply:
see updated date for boot_time